config/src/main/java/org/springframework/security/config/annotation/web/configurers/CorsConfigurer.java

```diff
@@ -98,24 +99,20 @@ private CorsFilter getCorsFilter(ApplicationContext context) {
 
 	static class MvcCorsFilter {
 
-		private static final String HANDLER_MAPPING_INTROSPECTOR_BEAN_NAME = "mvcHandlerMappingIntrospector";
-
 		/**
 		 * This needs to be isolated into a separate class as Spring MVC is an optional
 		 * dependency and will potentially cause ClassLoading issues
 		 * @param context
 		 * @return
 		 */
 		private static CorsFilter getMvcCorsFilter(ApplicationContext context) {
-			if (!context.containsBean(HANDLER_MAPPING_INTROSPECTOR_BEAN_NAME)) {
-				throw new NoSuchBeanDefinitionException(HANDLER_MAPPING_INTROSPECTOR_BEAN_NAME, "A Bean named "
-						+ HANDLER_MAPPING_INTROSPECTOR_BEAN_NAME + " of type "
-						+ HandlerMappingIntrospector.class.getName()
-						+ " is required to use MvcRequestMatcher. Please ensure Spring Security & Spring MVC are configured in a shared ApplicationContext.");
+			if (context.getBeanNamesForType(CorsConfigurationSource.class).length > 0) {
```

Comment:
We still need to look need to look it up by name in order to remain passive. Otherwise, if there are multiple in the application context, this logic will throw an error where before it didn't.

Reply:
Hi, @jzheaux, there's no method in `ApplicationContext` which checks that a bean with specific name and type exists, should we check it using `context.getBeanNamesForType(CorsConfigurationSource.class)` and then find the one with name `mvcHandlerMappingIntrospector` and type `CorsConfigurationSource`?